### PR TITLE
CSRF Protection

### LINKED ISSUE #213: Implement CSRF
- [ ] find out how this will work with the react router
- [ ] should do csrf on per-request basis (rather than per-page-load)?
 - is this really more secure?
 - hash per request or secret per request?

### PR DESCRIPTION
**Pull request by cadecairos**
Fixes #213

### CODE REVIEW

**Comment by cadecairos** [comment]
R? @jbuck and/or @simonwex

**Comment by cadecairos** [comment]
@jbuck crumb does not provide an API for regenerating the crumb manually, AFAICT

**Comment by simonwex** [comment]
@cadecairos I think an automated test here should be pretty easy to write and pretty important to catch regressions if we screw with headers in the future.

**Comment by cadecairos** [comment]
@simonwex Added two tests for each POST request endpoint and a test to check that the crumb cookie is set when a GET request is received

File: web/server.js [hunk 1]
```diff
@@ -45,6 +45,20 @@ module.exports = function(options) {
     });
   });
 
+  function skipCSRF(request, reply) {
+    return reply(true);
```
Comment (jbuck) [inline web/server.js]: Is this correct? From the docs:

```
'skip' - a function with the signature of function (request, reply) {}, which when provided, is called for every request. If the provided function returns true, validation and generation of crumb is skipped (defaults to false)
```

I think you just need to `return true` here?

Comment (jbuck) [inline web/server.js]: Yeah, looking at the code, this needs to be fixed up: https://github.com/hapijs/crumb/blob/master/lib/index.js#L62

**Comment by jbuck** [comment]
@cadecairos r+

**Pull request merged**